### ISSUE TITLE
Error : Expected path to be a path.

### ISSUE THREAD

**Issue by cmkpl** [neo4j-contrib/neo4j-apoc-procedures#20]
```
MATCH (a:STATION {name: 'A'}), (b:STATION{name: 'D'})
CALL apoc.algo.dijkstra(a, b, 'ROUTE', 'cost') yield path as path, weight as weight
RETURN path, weight, rels(path)
```

It will return error:

> Expected path to be a path.

Is it true that "path" is not in path data type?

**Comment by tkroman** [comment]
This is likely a bug in the `ProcedureCallPipe`, which uses 
`org.neo4j.cypher.internal.compiler.v3_0.helpers.ScalaCompatibility#asScalaCompatible` to convert Java types to Scala, and you can see that there's this line:

`case c: JavaIterable[_] => c.asScala.map(asScalaCompatible)`

which, in combination with the fact that paths are `Iterable`s, results in `path` being returned as `::` to the client.
I don't think this is worth working around in this project (apoc). Perhaps you can file this to the main (neo4j/neo4j) issue tracker.

**Comment by cmkpl** [comment]
Submitted there. Thanks.

**Comment by cmkpl** [comment]
I am just curious, is there any workaround I can do at my own set of code at the moment?

**Comment by boggle** [comment]
@cmkpl I'm afraid, don't think so; this is being fixed soonish though.

**Comment by tkroman** [comment]
@jexp This one should be closed with the 3.0.2 out there.
